// Copyright 2022 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;
option java_package = "org.chromium.chrome.browser.assistant_ui.proto";
option java_multiple_files = true;

package autofill_assistant.assistant_ui;

message AssistantUiAction {
  message AutofillAssistantUiConfiguration {
    optional bool hide_ui = 1;
  }

  message ContinueImmediately {}

  message BlockUntilUserAction {
    // If true, it will also continue when a DOM condition matches. Enabling
    // this is required for interrupts to run too.
    optional bool check_dom_conditions = 1;

    // Action will finish after this many ms, even if no user action was
    // performed.
    optional int32 timeout_ms = 2;
  }

  // This is an encoded AutofillAssistantUiState.
  optional bytes update_ui = 1;
  optional AutofillAssistantUiConfiguration configuration = 2;

  oneof continue_mode {
    ContinueImmediately continue_immediately = 3;
    BlockUntilUserAction block_until_user_action = 4;
  }
}

message AssistantUiActionResult {
  message DomConditionList {
    // This is the id of the conditions that matched, from
    // ExternalActionProto#ExternalCondition.
    repeated int32 condition_id = 1;
  }

  oneof response_type {
    // The action didn't wait for any user action.
    bool immediate = 1;
    // There was a timeout waiting for user action.
    bool timeout = 2;
    // The user selected a chip.
    string selected_chip = 3;
    // The external action finished because a DOM condition was satisfied.
    DomConditionList dom_conditions = 4;
  }
}
